src/components/DataTable/tools/instanceId.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * Generic utility to initialize a method that will return a unique instance id
+ * for a component.
+ */
+const setupGetInstanceId = () => {
```

Comment:
Great to have this code used library-wide, not just for table!

Reply:
Good call, I think there is `src/tools/uniqueId` but it initializes the id variable in the module scope. We should definitely update that one to follow this pattern so that the value isn't shared across modules 😄